index.js

```diff
@@ -24,6 +25,17 @@ module.exports = function(dep, filename, directory) {
 
   if (!depExt) {
     filepath += fileExt;
+  } else {
+    // If a dependency starts with a period AND it doesn't already end
+    // in .js AND doesn't use a custom plugin, add .js back to path.
+    if (fileExt === '.js' && !dep.endsWith('.js') && dep.indexOf('!') < 0) {
+      filepath += fileExt;
+    } else {
+      // If using a SystemJS style plugin
+      if (depExt.indexOf('!') > -1) {
+        filepath += depExt.substring(0, depExt.indexOf('!'));
```

Comment:
This is really the resolution of a systemjs plugin path. You're saying that the systemjs loader will see `foo.css!` and resolve it to `foo.css`. This hints at a larger point where maybe plugin resolution should happen in this module. 

I had a separate one for the amd case: https://github.com/mrjoelkemp/node-module-lookup-amd and used that and this in the order here https://github.com/mrjoelkemp/node-module-lookup-amd:

1. Lookup the path (i.e., resolve aliases or plugin loading)
2. Resolve the dependency's path (to an absolute path)

The idea of "resolving a dependency's path" incorporates both steps really. What would be awesome is to have a module that takes in a dependency's path and returns the absolute path of that module on the filesystem.

Custom resolvers like sass could be hooked in based on the extension of the given filepath.

Thoughts?

I'm inclined to merge this and then follow up with the above functionality. It would save me a lot of redundant logic across a few modules.